cmake/CompilerFeatureTest.cmake

```diff
@@ -0,0 +1,25 @@
+# # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Functions that determine compiler capabilities
+
+include(CheckCXXSourceCompiles)
+
+# Determines if the selected C++ compiler has ranges support.
+# Sets 'result_var' to whether support is detected.
+function(beman_check_range_support result_var)
+    # Check if the C++ standard is at least C++20 or later.
+    if(CMAKE_CXX_STANDARD LESS 20)
+        set(${result_var} FALSE PARENT_SCOPE)
+        return()
+    endif()
```

Comment:
`CMAKE_CXX_STANDARD` should not be used to determine if a compiler has C++20 support as it'll give both false negatives and false positives the way it is defined. You have to use `check_cxx_source_compiles` for specific features.